aya/src/util.rs

```diff
@@ -337,6 +337,18 @@ pub(crate) fn bytes_of_slice<T: Pod>(val: &[T]) -> &[u8] {
     unsafe { slice::from_raw_parts(val.as_ptr().cast(), size) }
 }
 
+pub(crate) fn bytes_of_c_char(val: &[std::os::raw::c_char]) -> &[u8] {
+    let length = val
```

Comment:
I'm alright with that, makes more sense to me I'm also adding a comment

```
    // length is the end of the C-style string (nul byte).
```

the the only difference is with your suggestion is that we default to `val.len()` but the C char should ALWAYS end with a NULL pointer so I'm going to leave it as `unwrap_or(0)`

Reply:
I don't think that's true. See the other discussion with @dave-tucker above: in the case where the slice is backed by an array, the string is allowed to take up the whole array without a NUL terminator.

Perhaps this method should take an array reference rather than a slice so that this is clearer.

`fn bytes_of_c_char<const N: usize>(vale: &[std::os::raw::c_char; N]) -> &[u8] { ... }`

Reply:
I haven't done this yet but if we can get some confirmation on the all c strings end in null thing I can do this 👍

Reply:
I don't understand what you're asking for. C is the wild west - strings usually terminate in NULL, but not always. I grepped around for BPF_OBJ_NAME_LEN in the kernel, and I think it *is* NULL-terminated (https://github.com/search?q=repo%3Atorvalds%2Flinux%20BPF_OBJ_NAME_LEN&type=code), so dave is probably right and this should return a CStr.

But we should make that change in another PR.